iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -97,58 +100,54 @@ public IotHubConnectionInfo(IotHubConnectionStringBuilder builder)
-            string moduleId = null,
-            string hostName = null,
-            Uri httpsEndpoint = null,
-            Uri amqpEndpoint = null,
-            string audience = null,
-            string sharedAccessKeyName = null,
-            string sharedAccessKey = null,
-            string sharedAccessSignature = null,
-            bool isUsingGateway = false)
-        {
-            IotHubName = iotHubName;
-            DeviceId = deviceId;
-            ModuleId = moduleId;
-            HostName = hostName;
-            HttpsEndpoint = httpsEndpoint;
-            AmqpEndpoint = amqpEndpoint;
-            Audience = audience;
-            SharedAccessKeyName = sharedAccessKeyName;
-            SharedAccessKey = sharedAccessKey;
-            SharedAccessSignature = sharedAccessSignature;
-            IsUsingGateway = isUsingGateway;
+            if (ClientOptions.TransportSettings.ClientCertificate == null)
+            {
+                AmqpCbsAudience = CreateAmqpCbsAudience();
+                AuthenticationModel = SharedAccessKeyName == null
+                    ? AuthenticationModel.SasIndividual
+                    : AuthenticationModel.SasGrouped;
+            }
+            else
+            {
+                AuthenticationModel = AuthenticationModel.X509;
+            }
         }
 
-        public AuthenticationWithTokenRefresh TokenRefresher { get; private set; }
+        public AuthenticationWithTokenRefresh TokenRefresher { get; }
+
+        public string IotHubName { get; }
+
+        public string DeviceId { get; }
 
-        public string IotHubName { get; private set; }
+        public string ModuleId { get; }
 
-        public string DeviceId { get; private set; }
+        public string HostName { get; }
 
-        public string ModuleId { get; private set; }
+        public string Audience { get; }
 
-        public string HostName { get; private set; }
+        public string SharedAccessKeyName { get; }
 
-        public Uri HttpsEndpoint { get; private set; }
+        public string SharedAccessKey { get; }
 
-        public Uri AmqpEndpoint { get; private set; }
+        public string SharedAccessSignature { get; }
 
-        public string Audience { get; private set; }
+        public bool IsUsingGateway { get; }
 
-        public string SharedAccessKeyName { get; private set; }
+        public AuthenticationModel AuthenticationModel { get; }
 
-        public string SharedAccessKey { get; private set; }
+        public IotHubClientOptions ClientOptions { get; }
 
-        public string SharedAccessSignature { get; private set; }
+        // TODO (abmisr): Consolidate with Audience
+        public string AmqpCbsAudience { get; }
 
-        public bool IsUsingGateway { get; private set; }
+        public bool IsPooling()
+        {
+
+            return AuthenticationModel != AuthenticationModel.X509
+                && ClientOptions.TransportSettings is IotHubClientAmqpSettings iotHubClientAmqpSettings1
```

Comment:
Why iotHubClientAmqpSettings**1**?

Reply:
Ah, I was reworking the logic, and this was to avoid conflict with the existing logic. Updated.